packages/fxa-settings/src/models/Session.ts

```diff
@@ -13,3 +25,93 @@ export const GET_SESSION_VERIFIED = gql`
     }
   }
 `;
+
+export const DESTROY_SESSION = gql`
+  mutation DestroySession {
+    destroySession(input: {}) {
+      clientMutationId
+    }
+  }
+`;
+
+export class Session implements SessionData {
+  private readonly authClient: AuthClient;
+  private readonly apolloClient: ApolloClient<object>;
+  private _loading: boolean;
+  token: string | null;
+  verified: boolean | null;
+
+  constructor(
+    authClient: AuthClient,
+    apolloClient: ApolloClient<NormalizedCacheObject>
+  ) {
+    this.authClient = authClient;
+    this.apolloClient = apolloClient;
+    this._loading = false;
+
+    this.token = sessionToken();
+    this.verified = null;
+  }
+
+  private async withLoadingStatus<T>(promise: Promise<T>) {
+    this._loading = true;
+    try {
+      return await promise;
+    } catch (e) {
+      throw e;
+    } finally {
+      this._loading = false;
+    }
+  }
+
+  async verifySession(
+    code: string,
+    options: {
+      service?: string;
+      scopes?: string[];
+      marketingOptIn?: boolean;
+      newsletters?: string[];
+    } = {}
+  ) {
+    await this.withLoadingStatus(
+      this.authClient.sessionVerifyCode(this.token!, code, options)
```

Comment:
`this.token!` is failing with an `unexpected error` until the code expires and a new out is requested. Replacing `this.token!` with a direct call to `sessionToken()!` appears to resolve the issue.